open-api/rest-catalog-open-api.yaml

```
@@ -1482,6 +1490,33 @@ components:
       explode: false
       example: "vended-credentials,remote-signing"
 
+    page-token:
+      name: pageToken
+      in: query
+      description:
+        An opaque token which allows clients to make use of pagination for a list API (e.g. ListTables) by signaling to the service that they would
+        that they would prefer response to be paginated. If pageSize is specified, the service should return exactly the number of items specified by it.
+        
+        Clients that support pagination initiate the request by sending an empty pageToken e.g. GET /tables?pageToken
+        Servers supporting pagination would recognize pagination is requested due to the presence of the pageToken
+        and honor the contracts specified above by returning a NextPageToken in response if there are more results available.
+        Servers that are non-pagination aware will ignore the token and return all results.
+      required: false
+      allowEmptyValue: true
+      schema:
+        type: string
+
+    page-size:
+      name: pageSize
+      in: query
+      description:
+        The number of results to return on the next page for servers that support pagination.
```

I think this is the **max** number of results right? Because you might return fewer if this is the last page, and also service might decide to return fewer results due to various reasons (e.g. payload too big)

---

Does this sound better? 

```suggestion
        For servers that support pagination, this signals an upper bound of the number of results that client will receive. 
        For servers that do not support pagination, clients may receive results larger than the indicated pageSize.
```